src/App.tsx

```diff
@@ -1,24 +1,208 @@
+      clearTimeout(timer);
+    };
+  }, [setIsHidden]);
+
+  useEffect(() => {
+    loadTodos();
+  }, []);
+
+  useEffect(() => {
+    setFilTodos(todos);
+  }, [todos]);
+
+  useEffect(() => {
+    const newTodos = todos.filter(todo => {
+      switch (filter) {
+        case Filter.Active:
+          return !todo.completed;
+
+        case Filter.Completed:
+          return todo.completed;
+
+        case Filter.All:
+        default:
+          return true;
+      }
+    });
+
+    setFilTodos(newTodos);
+  }, [todos, filter]);
+
+  useEffect(() => {
+    setTimeout(() => {
+      setNewTodoLoad(-1);
+    }, 3000);
+  }, [newTodoLoad]);
+
+  const handleUpdate = async (todo: Todo) => {
+    setProcessingTodoId(prev => [...prev, todo.id]);
+
+    try {
+      const updatedTodo = await updateTodo(todo.id, todo);
+
+      setTodos(prev => prev.map(prevTodo => (
+        prevTodo.id === updatedTodo.id
+          ? updatedTodo
+          : prevTodo
+      )));
+    } catch (error) {
+      setTodosError(ErrorMessage.UnableToUpdateTodo);
+
+      throw new Error();
+    } finally {
+      setProcessingTodoId(prev => prev.filter(id => id !== todo.id));
+    }
+  };
+
+  const toggleAll = async () => {
+    const isAllCompleted = todos.every(todo => todo.completed);
+
+    const todosToUpdate = todos.filter(todo => (isAllCompleted
+      ? todo.completed
+      : !todo.completed
+    ));
+
+    await Promise.all(todosToUpdate.map(todo => (
+      handleUpdate({
+        ...todo,
+        completed: !isAllCompleted,
+      })
+    )));
+  };
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">React Todo App - Add and Delete</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <TodoHeader
+          todos={filteredTodos}
+          todo={todos}
+          setTodos={setTodos}
+          setTodosError={setTodosError}
+          tempTodos={tempTodos}
+          setTempTodos={setTempTodos}
+          onTodoToggle={toggleAll}
+          setIsLoading={setIsLoading}
+          setNewTodoLoad={setNewTodoLoad}
+          updatedTodos={updatedTodos}
+        />
+
+        <section className="todoapp__main" data-cy="TodoList">
+          {
+            filteredTodos.map(todo => (
+              <TodoItem
+                todo={todo}
+                key={todo.id}
+                todos={filteredTodos}
+                setTodos={setTodos}
+                isLoading={isLoading}
+                setIsLoading={setIsLoading}
+                setTodosError={setTodosError}
+                onTodoUpdate={handleUpdate}
+                isNew={(todo.id === newTodoLoad) && true}
```

Comment:
```suggestion
                isNew={todo.id === newTodoLoad}
```

should work same but simplify, check all places